cli/stream_command.go

```diff
@@ -1861,14 +1851,38 @@ func (c *streamCmd) showStreamInfo(info *api.StreamInfo) {
 	showSource := func(s *api.StreamSourceInfo) {
 		cols.AddRow("Stream Name", s.Name)
 
-		if s.SubjectTransformDest != "" {
+		switch s.SubjectTransformDest {
+		case "":
+			switch len(s.SubjectTransforms) {
+			case 0:
+				cols.AddRowIfNotEmpty("Subject Filter", s.FilterSubject)
+			default:
+				for i := range s.SubjectTransforms {
+					t := ""
+
+					if i == 0 {
+						if len(s.SubjectTransforms) > 1 {
+							t = "Subject Filters and Transforms"
+						} else {
+							t = "Subject Filter and Transform"
+						}
+					}
+
+					if s.SubjectTransforms[i].Destination == "" {
+						cols.AddRowf(t, "%s to [no transform]", s.SubjectTransforms[i].Source)
+					} else {
+						cols.AddRowf(t, "%s to %s", s.SubjectTransforms[i].Source, s.SubjectTransforms[i].Destination)
+					}
+				}
+			}
+		default:
```

Comment:
purpose of using switch is to remove nested conditionals they evaluate top-down and so great for removing this kind of nested logic

```
switch {
case s.SubjectTransformDest != "": 
  // outer switch default clause

case len(s.SubjectTransforms) == 0:
  // inner switch case 0 clause

case len(s.SubjectTransforms) > 0:
 // inner switch default clause

}```

Reply:
I was actually trying to represent the nested aspect of the logic (i.e. is SubjectTransformDest == "" then look at SubjectTransforms and depending on it's length do this or that) because that's how I represent what I want to do in my head, hence the if/then logic but fair enough maybe that makes it harder for someone else who doesn't know exactly what I want the code to do to follow.

Flattened it to a single switch which is a few lines shorter anyways (and I always like less lines of code)

Reply:
I definitely take clarity over lines of code :) but yeah it’s just a thing with idiomatic go to try and not nest logic so it always gets me